### PR TITLE
Nightly updates

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: src/rfc2047.rs [hunk 1]
```diff
@@ -40,38 +39,34 @@ pub fn decode_rfc2047(s: &str) -> Option<String> {
...
+    loop {
+        match char_iter.next() {
+            Some('=') => {
                 let mut hex_string = String::new();
                 for _ in 0..2 {
-                    let hex_digit_char = s.char_range_at(inner_pos);
-                    hex_string.push(hex_digit_char.ch);
-                    inner_pos = hex_digit_char.next;
-                }
+                    let hex_digit_char = char_iter.next().unwrap();  // FIXME
```
Comment (untitaker) [inline src/rfc2047.rs]: I find this highly problematic but am unsure if and how to fix. I have no idea how the previous code even behaved if it hit string boundaries?

File: src/rfc5322.rs [hunk 2]
```diff
@@ -152,7 +151,7 @@ impl<'s> Rfc5322Parser<'s> {
     }
 
     /// Consume an unstructured from the input.
-    #[unstable]
+    /// [unstable]
```
Comment (untitaker) [inline src/rfc5322.rs]: unstable-markers are deprecated, so I converted them to docstrings to keep the info (for API review).

File: src/rfc5322.rs [hunk 3]
```diff
@@ -325,26 +324,26 @@ impl<'s> Rfc5322Parser<'s> {
...
     pub fn consume_char(&mut self) -> char {
         if self.eof() { 
             // TODO: Consider making this return an Option<char>
             panic!("Consuming beyond end of input");
         }
-        let ch_range = self.s.char_range_at(self.pos);
-        self.pos = ch_range.next;
-        ch_range.ch
+        let c = self.s.char_at(self.pos);
```
Comment (untitaker) [inline src/rfc5322.rs]: `char_at` is also unstable, but the phrasing of the error message suggests it's more likely to become stable -- char_range_at, however, seems to be discouraged.

File: src/rfc822.rs [hunk 4]
```diff
@@ -147,14 +147,14 @@ impl<'s> Rfc822DateParser<'s> {
...
-    #[unstable]
+    /// [unstable]
     pub fn consume_datetime(&mut self) -> ParsingResult<DateTime<FixedOffset>> {
         // Handle the optional day ","
         self.parser.push_position();
         let day_of_week = self.parser.consume_word(false);
         if day_of_week.is_some() {
             let lower_dow = day_of_week.unwrap().into_ascii_lowercase();
-            if DAYS_OF_WEEK.position_elem(&lower_dow.as_slice()).is_some() {
+            if DAYS_OF_WEEK.position_elem(&&lower_dow[..]).is_some() {
```
Comment (untitaker) [inline src/rfc822.rs]: Why are double-pointers necessary here?

File: Cargo.toml [hunk 5]
```diff
@@ -9,9 +9,9 @@ documentation = "https://niax.github.io/rust-email/email/"
 license = "MIT"
 
 [dependencies]
-encoding = "~0.2.12"
-time = "~0.1.7"
-chrono = "~0.2.1"
-lazy_static = "~0.1.4"
-rustc-serialize = "0.2"
-rand = "~0.1.2"
+encoding = "0.2"
```
Comment (untitaker) [inline Cargo.toml]: I'd suggest that we just use ``"*"`` here.

**Comment by untitaker** [comment]
@niax Could you take a look at this?

It also seems that you don't have the time anymore to maintain this library -- I'd be glad to do so.

**Comment by niax** [comment]
You're right, I recently switched jobs and now don't have the time to maintain this. I'll look in to how to transfer ownership on the crate.

**Comment by niax** [comment]
@untitaker you should now have ownership over the crate on crates.io

Thanks for your continued maintenance.

**Comment by untitaker** [comment]
Should I fork this repo or could I get commit rights for this one?

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**